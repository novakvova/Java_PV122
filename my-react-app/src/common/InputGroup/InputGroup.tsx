import {FC, InputHTMLAttributes} from "react";
import * as classNames from "classnames";

interface InputGroupProps extends InputHTMLAttributes<HTMLInputElement> {
    field: string;
    label: string;
    type?: "text"|"email"|"password"|"number";
    placeholder?: string;
    touched?: boolean;
    error?: string;
}
const InputGroup: FC<InputGroupProps> = ({
                                             field,
                                             label,
                                             type="text",
                                             placeholder,
                                             error,
                                             touched,
                                             onChange,
                                             onBlur,
                                             value
                                         }) => {

    const isError = error && touched;
    return (
        <>
            <div className="mb-4">
                <label className={classNames("mb-2.5 block font-medium",
                    {"text-black dark:text-white": !isError},
                    {"text-red-700 dark:dark:text-red-500": isError}
                )}
                       htmlFor={field}
                >
                    {label}
                </label>
                <div className="relative">
                    <input
                        id={field}
                        name={field}
                        type={type}
                        value={value}
                        onChange={onChange}
                        onBlur={onBlur}
                        placeholder={placeholder}
                        className={classNames("w-full rounded-lg border bg-transparent py-4 pl-6 pr-10 outline-none focus:border-primary focus-visible:shadow-none dark:bg-form-input",
                            {"border-stroke dark:border-form-strokedark ": !isError},
                            {"border-red-500 text-red-900 placeholder-red-700 dark:border-r-form-strokedark ": isError })}
                    />
                </div>
                {isError &&
                    <p className="mt-2 text-sm text-red-600 dark:text-red-500">{error}</p>
                }

            </div>
        </>
    )
}

export default InputGroup;