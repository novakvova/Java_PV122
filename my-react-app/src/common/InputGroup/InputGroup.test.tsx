import '@testing-library/jest-dom';
import {render, screen} from "@testing-library/react";
import InputGroup from "./index.ts";

describe('InputGroup Conponent TESTS', ()=> {

    test("Test Label props component", () => {
        render(<InputGroup field={"ssdf"} label={"Login"} onClick={()=>{}} />);
        expect(screen.getByText("Login")).toBeInTheDocument();
    });

    it("InputGroup smapshot", ()=> {
       const ig = render(
           <InputGroup
               field={"email"}
               label={"Електронна пошта"}
               type={"email"}
               placeholder={"Вкажіть пошту"}
               onChange={()=>{}} />
       );
       expect(ig).toMatchSnapshot();
    });

});
