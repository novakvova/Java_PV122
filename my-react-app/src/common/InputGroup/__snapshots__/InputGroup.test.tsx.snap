// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup Conponent TESTS InputGroup smapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-4"
      >
        <label
          class="mb-2.5 block font-medium text-black dark:text-white"
          for="email"
        >
          Електронна пошта
        </label>
        <div
          class="relative"
        >
          <input
            class="w-full rounded-lg border bg-transparent py-4 pl-6 pr-10 outline-none focus:border-primary focus-visible:shadow-none dark:bg-form-input border-stroke dark:border-form-strokedark "
            id="email"
            name="email"
            placeholder="Вкажіть пошту"
            type="email"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-4"
    >
      <label
        class="mb-2.5 block font-medium text-black dark:text-white"
        for="email"
      >
        Електронна пошта
      </label>
      <div
        class="relative"
      >
        <input
          class="w-full rounded-lg border bg-transparent py-4 pl-6 pr-10 outline-none focus:border-primary focus-visible:shadow-none dark:bg-form-input border-stroke dark:border-form-strokedark "
          id="email"
          name="email"
          placeholder="Вкажіть пошту"
          type="email"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
